src/client/OpsService.ts

```diff
@@ -146,19 +146,45 @@ export class OpsService {
 	 * const result = await client.read(filter)
 	 * ```
 	 *
+	 * You can also read by ids by passing a list of Refs or strings as the first arguement
+	 * ```typescript
+	 * const result = await client.read(['@id1', '@id2', '@id3'])
+	 * ```
+	 *
 	 * https://project-haystack.org/doc/Ops#read
 	 *
 	 * @param filter The required haystack filter.
 	 * @param limit Optional limit on the number of records sent back.
 	 * @returns A grid with the resolved query.
 	 */
-	public async read(filter: string, limit?: number): Promise<HGrid> {
-		return this.invokeOp(
-			'read',
-			HDict.make(
-				typeof limit === 'number' ? { filter, limit } : { filter }
+	public async read(ids: (string | HRef)[], limit?: number): Promise<HGrid>
+	public async read(filter: string, limit?: number): Promise<HGrid>
+	public async read(params: unknown, limit?: number): Promise<HGrid> {
```

Comment:
- The filter typedoc comment needs updating. 
- The limit comment needs updating so the caller understands when it's not used.
- Use strongly typed parameters instead of unknown.
- Prefer `Array.isArray` over instance of Array.
- HRef already handles the Zinc encoding for you. You can also pass in HRef references that just return themselves.

```
	/**
	 * Reads records from the server.
	 *
	 * Please note, to help build a Haystack filter you can use HFilterBuilder available in
	 * Haystack Core.
	 *
	 * ```typescript
	 * const filter = new HFilterBuilder()
	 *	.has('site')
	 *	.and()
	 *	.equals('geoCity', 'Las Vegas')
	 *	.build()
	 *
	 * const result = await client.read(filter)
	 * ```
	 *
	 * You can also read by ids by passing a list of Refs or strings as the first arguement
	 * ```typescript
	 * const result = await client.read(['id1', 'id2', 'id3'])
	 * ```
	 *
	 * https://project-haystack.org/doc/Ops#read
	 *
	 * @param params a filter string or an array or refs.
	 * @param limit Optional limit on the number of records sent back when a filter is used.
	 * @returns A grid with the resolved query.
	 */
	public async read(
		params: string | string[] | HRef[],
		limit?: number
	): Promise<HGrid> {
		let requestGrid: HGrid | undefined

		if (Array.isArray(params)) {
			requestGrid = HGrid.make(
				params.map((param) =>
					HDict.make({
						id: HRef.make(param),
					})
				)
			)
		} else if (typeof params === 'string') {
			requestGrid = HDict.make(
				typeof limit === 'number'
					? { filter: params, limit }
					: { filter: params }
			).toGrid()
		}

		return requestGrid
			? this.invokeOp('read', requestGrid)
			: Promise.reject(new Error('Invalid Request'))
	}

Reply:
Looking at typedoc, I'll break the comments up into two. Each overload should have their own doc comment.